Python/flowgraph.c

```diff
@@ -1384,6 +1393,62 @@ fold_tuple_on_constants(PyObject *const_cache,
     return SUCCESS;
 }
 
+// Replace LOAD_CONST x, LOAD_CONST y, LOAD_CONST z, BUILD_LIST 3
+// with BUILD_LIST 0, LOAD_CONST (x, y, z), LIST_EXTEND 1
+// or BUILD_SET & SET_UPDATE respectively.
+static int
+optimize_const_sequence(PyObject *const_cache,
+                        cfg_instr* inst,
+                        int n, PyObject *consts,
+                        int build, int extend)
+{
+    assert(PyDict_CheckExact(const_cache));
+    assert(PyList_CheckExact(consts));
+    assert(inst[n].i_oparg == n);
+    int list = build == BUILD_LIST;
+    if (list) {
+        assert(inst[n].i_opcode == BUILD_LIST);
+    }
+    else {
+        assert(inst[n].i_opcode == BUILD_SET);
+    }
```

Comment:
Looks like the `build` arg is redundant as well. It is equal to `inst[n].i_opcode`.

Also, I found the `list` variable name confusing - I expected it to be a list object. Maybe `construct_list` (if we need it at all).

Reply:
Also `extend` can be deduced from `build`, so not really necessary.